### PR TITLE
improved an error message for missing sequence dictionary. Closes #1305.

### LINKED ISSUE #1305: improve message or missing seq dict on variants
`./gatk-launch CountVariants --variant  some.vcf -L 21`

i get

`A USER ERROR has occurred: We currently require a sequence dictionary (from a reference or source of reads) to process intervals. This restriction may be removed in the future.`

the message should tell me what to do. References to the future are less useful than help in what argument to pass

### PR DESCRIPTION
**Pull request by davidbenjamin**
@droazen closes an old issue.

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java [hunk 1]
```diff
@@ -322,8 +322,12 @@ void initializeIntervals() {
...
             if ( sequenceDictionary == null ) {
-                throw new UserException("We currently require a sequence dictionary (from a reference, a source of reads, or a source of variants) " +
-                                        "to process intervals. This restriction may be removed in the future.");
+                throw new UserException("We require a sequence dictionary from a reference, a source of reads, or a source of variants to process intervals.  " +
+                        "A sequence dictionary is a list of header lines of contig names, lengths, and assemblies.  " +
+                        "Since reference and reads files generally contain sequence dictionaries, this error most commonly occurs " +
+                        "for VariantWalkers that do not require a reference or reads.  You can fix the problem by passing a reference file with a sequence dictionary " +
+                        "via the -R argument or you can paste lines of the form: ##contig=<ID=20,length=63025520,assembly=GRCh37> (one line per contig) " +
+                        "into your vcf header and then re-indexing your vcf with the IndexFeatureFile GATK tool.");
```
Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java]: re-indexing -> reindex?

**Comment by samuelklee** [comment]
Also, I think if you put "Closes ###" in the commit message, the issue is not automatically closed by GitHub.

**Comment by samuelklee** [comment]
By the way, thanks for going through and cleaning up old issues, @davidbenjamin!

File: src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java (hunk 1 above)
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java]: done

File: src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java [hunk 2]
```diff
@@ -322,8 +322,12 @@ void initializeIntervals() {
...
             final SAMSequenceDictionary sequenceDictionary = getBestAvailableSequenceDictionary();
             if ( sequenceDictionary == null ) {
-                throw new UserException("We currently require a sequence dictionary (from a reference, a source of reads, or a source of variants) " +
-                                        "to process intervals. This restriction may be removed in the future.");
+                throw new UserException("We require a sequence dictionary from a reference, a source of reads, or a source of variants to process intervals.  " +
+                        "A sequence dictionary is a list of header lines of contig names, lengths, and assemblies.  " +
+                        "Since reference and reads files generally contain sequence dictionaries, this error most commonly occurs " +
+                        "for VariantWalkers that do not require a reference or reads.  You can fix the problem by passing a reference file with a sequence dictionary " +
+                        "via the -R argument or you can paste lines of the form: ##contig=<ID=20,length=63025520,assembly=GRCh37> (one line per contig) " +
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java]: I really don't think we should be recommending manually editing vcf files in a text editor.

Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java]: replaced with advice to use the `UpdateVCFSequenceDictionary` tool.

**Comment by droazen** [comment]
:+1: looks good -- merging

**Pull request merged**